OneCommunityGlobal/HGNRest
# Roberto - For Main Branch- sends email to Jae whenever Admin or Owner accounts are created or password reset

# Description
<img width="819" alt="Screenshot 2023-10-10 at 1 34 00 PM" src="https://github.com/OneCommunityGlobal/HGNRest/assets/117053202/27f16c0b-06c2-457c-ab8a-f330c38e1ae4">
Like PR 569 this PR sends an email whenever an Owner account is created or their password is reset. However this PR is meant for the main branch so it also includes an email sent for the creation of Admin accounts and their passwords reset as well.

## Related PRS (if any):
This backed PR is a continuation of [569](https://github.com/OneCommunityGlobal/HGNRest/pull/569)

## Main changes explained:
- Imports and uses emailSender function in userProfileController.js. Email is sent when a new owner account or admin account is created or password is reset. 
- Adds more security checks to the reset password function because it didn't originally check permissions.

## How to test:
1. Follow PR [569](https://github.com/OneCommunityGlobal/HGNRest/pull/569) instructions and set up first
2. Additionally verify that creating a new Admin account results in an email.
3. Verify that resetting an Admin's account results in an email as well.

## Screenshots or videos of changes:

## Note:
PR 569 is a prerequisite to this PR so check it out first. This is a continuation of that PR.
